cvmfs/cache_extern.cc

```diff
@@ -677,7 +691,7 @@ bool ExternalCacheManager::SpawnPlugin(const vector<string> &cmd_line) {
   retval = ManagedExec(cmd_line,
                        preserve_filedes,
                        map_fildes,
-                       false,  // drop_credentials
+                       true,   // drop_credentials
```

Comment:
Like the fuse module, at the moment plugins are effectively run by the cvmfs user but they can regain root persona.  Depending on the security settings of the distribution, regaining root is necessary to have the watchdog successfully connect gdb to the main process for a stack trace.  We could instead also let the plugin decide if it wants to give up on its hidden superpowers.  Or do it automatically in the libcvmfs_cache library after the watchdog has been spawned or when the plugins switches to listen mode.

Reply:
Is the watchdog really going to connect to this process though?  I thought the extern processes (like the authz one) is outside the purview here.

Reply:
The fuse processes watchdog is out of the loop but a plugin can spawn its own watchdog with `cvmcache_spawn_watchdog()`.

Reply:
@bbockelm I merged all but this change into the 2.4 branch, which will be merged into the devel branch.